src/serializer/ResidualHeapInspector.js

```diff
@@ -156,6 +156,7 @@ export class ResidualHeapInspector {
           desc.value !== undefined &&
           !this.realm.isCompatibleWith(this.realm.MOBILE_JSC_VERSION) &&
           !this.realm.isCompatibleWith("mobile") &&
+          desc.writable === (val instanceof FunctionValue && !val.$Strict && targetDescriptor.writable) &&
```

Comment:
We already know that `val instanceof FunctionValue` (outer `if`).
I don't understand why strictness affects the default value of a function's name.